compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -755,6 +755,7 @@ fn ty_is_local_constructor(tcx: TyCtxt<'_>, ty: Ty<'_>, in_crate: InCrate) -> bo
 
         ty::Adt(def, _) => def_id_is_local(def.did(), in_crate),
         ty::Foreign(did) => def_id_is_local(did, in_crate),
+        ty::TyAlias(did, _) => def_id_is_local(did, in_crate),
```

Comment:
hmm.... this is probably not right. should probably recurse onto the aliased type

Reply:
Ok, adding a `type_of`!